lib/rules/no-unused-vars.js

```diff
@@ -199,43 +247,21 @@ module.exports = {
             FunctionExpression: markFunctionReturnTypeAsUsed,
             ArrowFunctionExpression: markFunctionReturnTypeAsUsed,
 
-            ClassProperty: markDecoratorsAsUsed,
-            ClassDeclaration(node) {
-                markDecoratorsAsUsed(node);
-                markImplementedInterfacesAsUsed(node);
+            TSInterfaceHeritage(node) {
+                markExtendedInterfaceAsUsed(node);
+            },
+
+            Decorator(node) {
+                markDecoratorAsUsed(node);
             },
```

Comment:
Why isn’t this just `Decorator: markDecoratorAsUsed`?

Reply:
No reason, it was late last night 😄 
I will update shortly, TSInterfaceHeritage has the same problem...